crates/tools/msvc/src/main.rs

```diff
@@ -40,6 +42,31 @@ fn main() {
     for library in libraries.keys() {
         std::fs::remove_file(output.join(format!("{}.lib", library))).unwrap();
     }
+
+    // Clear out timestamps in the resulting library.
+    let mut archive = std::fs::File::options().read(true).write(true).open(output.join("windows.lib")).unwrap();
+    let len = archive.metadata().unwrap().len();
+    let mut header = [0u8; 8];
+    archive.read_exact(&mut header).unwrap();
+    assert_eq!(&header, b"!<arch>\n");
+    for num in 1.. {
+        archive.seek(SeekFrom::Current(16)).unwrap(); // identifier
+        assert_eq!(archive.write(b"0           ").unwrap(), 12); // replace archive timestamp
```

Comment:
To align with the Windows SDK (e.g. kernel32.lib), let's set this to `-1`.

```suggestion
        assert_eq!(archive.write(b"-1          ").unwrap(), 12); // replace archive timestamp
```

The below timestamp replacement will also need a tweak.

Reply:
Interestingly kernel32.lib has random timestamps for the member themselves. That said, does it really matter to set this to -1? This introduces a new difference with the files that llvm-dlltool produces for no benefit.

Reply:
I see `-1` for members in kernel32.lib from 22621.0. Looks like this is the sentinel value chosen by msvc and friends in the SDK, so we don't want to stray and cause problems. 

`dumpbin`, for example, already behaves differently when using `-1` and `0`:

```
// Timestamp set to 0
Archive member name at 25EBE: KERNEL32.dll/
       0 time/date Wed Dec 31 16:00:00 1969
         uid
         gid
       0 mode
     1F2 size
correct header end

// Timestamp set to -1
Archive member name at 25EBE: KERNEL32.dll/
FFFFFFFF time/date
         uid
         gid
       0 mode
     1F2 size
correct header end
```

Reply:
That's the timestamp for from the ar headers, and it doesn't matter. The timestamps in the members themselves also don't matter, and they are random.
```
Archive member name at 25EBE: KERNEL32.dll/
FFFFFFFF time/date
         uid
         gid
       0 mode
     1F2 size
correct header end

FILE HEADER VALUES
            8664 machine (x64)
               3 number of sections
        FAC6008A time date stamp
             10E file pointer to symbol table
               8 number of symbols
               0 size of optional header
               0 characteristics

Archive member name at 260EC: KERNEL32.dll/   
FFFFFFFF time/date
         uid
         gid
       0 mode
      FB size
correct header end

FILE HEADER VALUES
            8664 machine (x64)
               2 number of sections
        D33470E4 time date stamp
              BA file pointer to symbol table
               2 number of symbols
               0 size of optional header
               0 characteristics
(...)

Archive member name at 26380: KERNEL32.dll/
FFFFFFFF time/date
         uid
         gid
       0 mode
      39 size
correct header end

  Version      : 0
  Machine      : 8664 (x64)
  TimeDateStamp: D1A71C8A
  SizeOfData   : 00000025
  DLL name     : KERNEL32.dll
  Symbol name  : AcquireSRWLockExclusive
  Type         : code
  Name type    : name
  Hint         : 0
  Name         : AcquireSRWLockExclusive
```
(that's from SDK 22621)

Reply:
Ah yup, I see that now too thanks!

Given the demonstrable impact to `dumpbin`, I'm still leaning towards `-1` for the archive headers.

For the members though, we'll have to double-check with the msvc folks. We don't have enough information right now to know what "doesn't matter" or is "random". (Microsoft, for example, relies on random-looking PE TimeDateStamp values for symbol/image lookup in its symbol server.) I'm happy with whatever there for this PR then we go back and fix it later if needed.

Reply:
Probably related to the random-looking timestamps: [Why are the module timestamps in Windows 10 so nonsensical?](https://devblogs.microsoft.com/oldnewthing/20180103-00/?p=97705).